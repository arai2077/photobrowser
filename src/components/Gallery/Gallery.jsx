import React from 'react';
import Thumbnail from './Thumbnail/Thumbnail.jsx';
import { useGalleryStore } from '../../GalleryContext.js';
import { useObserver } from 'mobx-react';

const Gallery = () => {
  const galleryStore = useGalleryStore();

  return useObserver(() => (
    <ul>
      {galleryStore.photos.map(photo => {
        <Thumbnail photo={photo} />
      })}
    </ul>
  ));
}

export default Gallery;