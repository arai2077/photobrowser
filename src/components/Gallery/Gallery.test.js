import React from "react";
import { shallow } from "enzyme";
import Gallery from "./Gallery";

describe("Gallery", () => {
  it("renders", () => {
    const component = shallow(<Gallery />);

    expect(component).toMatchSnapshot();
  });
});
