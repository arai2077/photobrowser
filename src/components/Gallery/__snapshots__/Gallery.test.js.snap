// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders 1`] = `
<Fragment>
  <Thumbnail />
  <Thumbnail />
  <Thumbnail />
</Fragment>
`;
